exporters/elasticsearch/include/opentelemetry/exporters/elasticsearch/es_log_exporter.h

```diff
@@ -0,0 +1,160 @@
+  bool console_debug_;
+
+  /**
+   * Constructor for the ElasticsearchExporterOptions. By default, the endpoint is
+   * localhost:9200/logs with a timeout of 30 seconds and disabled console debugging
+   * @param host The host of the Elasticsearch instance
+   * @param port The port of the Elasticsearch instance
+   * @param index The index/shard that the logs will be written to
+   * @param response_timeout The maximum time the exporter should wait after sending a request to
+   * Elasticsearch
+   * @param console_debug Print the status of the exporter methods in the console
+   */
+  ElasticsearchExporterOptions(std::string host     = "localhost",
+                               int port             = 9200,
+                               std::string index    = "logs",
+                               int response_timeout = 30,
+                               bool console_debug   = false)
+      : host_{host},
+        port_{port},
+        index_{index},
+        response_timeout_{response_timeout},
+        console_debug_{console_debug}
+  {}
+};
+
+/**
+ * The ElasticsearchLogExporter exports logs to Elasticsearch in JSON format
+ */
+class ElasticsearchLogExporter final : public sdklogs::LogExporter
+{
+public:
+  /**
+   * Create an ElasticsearchLogExporter with default exporter options.
+   */
+  ElasticsearchLogExporter();
+
+  /**
+   * Create an ElasticsearchLogExporter with user specified options.
+   * @param options An object containing the user's configuration options.
+   */
+  ElasticsearchLogExporter(const ElasticsearchExporterOptions &options);
+
+  /**
+   * Exports a vector of log records to the Elasticsearch instance. Guaranteed to return after a
+   * timeout specified from the options passed from the constructor.
+   * @param records A list of log records to send to Elasticsearch.
+   */
+  sdklogs::ExportResult Export(const nostd::span<std::unique_ptr<opentelemetry::logs::LogRecord>>
+                                   &records) noexcept override;
+
+  /**
+   * Shutdown this exporter.
+   * @param timeout The maximum time to wait for the shutdown method to return
+   */
+  bool Shutdown(std::chrono::microseconds timeout = std::chrono::microseconds(0)) noexcept override;
+
+private:
+  // Stores if this exporter had its Shutdown() method called
+  bool isShutdown_ = false;
+
+  // Configuration options for the exporter
+  ElasticsearchExporterOptions options_;
+
+  // Object that stores the HTTP sessions that have been created
+  std::unique_ptr<ext::http::client::SessionManager> session_manager_;
+
+  /**
+   * Converts a log record into a nlohmann::json object.
+   */
+  json RecordToJSON(std::unique_ptr<opentelemetry::logs::LogRecord> record);
+
+  /**
+   * Converts a common::AttributeValue into a string, which is used for parsing the attributes
+   * and resource KeyValueIterables
+   */
+  const std::string ValueToString(const common::AttributeValue &value)
```

Comment:
same here, not sure if every single type should be converted to a string, or if original data types should be preserved when converted to json

Reply:
yup agreed, I'll take a look at switching it over to use the correct data types